src/components/Devnet/Devnet.tsx

```diff
@@ -1,43 +1,73 @@
 import { useEffect, useState } from 'react';
 import { getInstance } from '../../fhevmjs';
-
 import './Devnet.css';
 import { Eip1193Provider, getAddress } from 'ethers';
 
-const toHexString = (bytes: Uint8Array) => bytes.reduce((str, byte) => str + byte.toString(16).padStart(2, '0'), '');
+const toHexString = (bytes: Uint8Array) =>
+  bytes.reduce((str, byte) => str + byte.toString(16).padStart(2, '0'), '');
+
+export type DevnetProps = {
+  account: string;
+  provider: Eip1193Provider;
+};
 
-export type DevnetProps = { account: string; provider: Eip1193Provider };
+const CONTRACT_ADDRESS = getAddress(
+  '0x309cf2aae85ad8a1db70ca88cfd4225bf17a7482',
+);
 
 export const Devnet = ({ account, provider }: DevnetProps) => {
   const [handles, setHandles] = useState<Uint8Array[]>([]);
   const [encryption, setEncryption] = useState<Uint8Array>();
-  const [eip712, setEip712] = useState<ReturnType<typeof instance.createEIP712>>();
+  const [eip712, setEip712] =
+    useState<ReturnType<typeof instance.createEIP712>>();
+  const [walletAddress, setWalletAddress] = useState<string>('');
   const instance = getInstance();
 
+  // Handle wallet connection
+  useEffect(() => {
```

Comment:
account (and ethers provider) is already given as props `<Devnet account={account} provider={provider} />`

Reply:
Ok I removed the account setter in my new commit, sorry for the oversight